File: JanperCab/5-JanperCab/ClientApp/src/app/_models/delivery-sheet/ShippingSheetDto.ts
import { TruckDto } from './../truck/TruckDto';
import { DriverDto } from './../driver/DriverDto';
import { DeliverySheetDto } from './DeliverySheetDto';
import { DeliveryPatchDto } from './DeliveryPatchDto';
import { Type } from 'class-transformer';
import { DeliveryMethodEnum } from 'src/app/_enums/DeliveryMethodEnum';

export class ShippingSheetDto extends DeliverySheetDto {
  static readonly BARCODE_PREFIX: string = 'RS';

  driverId: number;
  truckId: number;

  @Type(() => DriverDto)
  driver: DriverDto;

  @Type(() => TruckDto)
  truck: TruckDto;

  constructor() {
    super();
    this.$type =
      '_3_Application.Dtos.DeliverySheet.ShippingSheetDto, 3-Application';
    this.deliveryMethod = DeliveryMethodEnum.Shipping;
  }

  getBarcodePrefix() {
    return ShippingSheetDto.BARCODE_PREFIX;
  }

  getDescription(): string {
    return `${this.driver.fullName} - ${this.truck.plateNumber}`;
  }

  getPatchDetails = (): DeliveryPatchDto[] => {
    const patches: DeliveryPatchDto[] = [];

    this.enquiriesForSheet.forEach((enquiry) => {
      const existPatch = patches.find((p) => p.hasSameAddress(enquiry));

      if (existPatch) {
        existPatch.enquiriesForSheet.push(enquiry);
      } else {
        patches.push(new DeliveryPatchDto(enquiry));
      }
    });

    return patches;
  };
}

File: JanperCab/3-Application/Dtos/DeliverySheet/ShippingSheetDto.cs
using _1_Domain.Enum;
using _3_Application.Dtos.Driver;
using _3_Application.Dtos.Truck;

namespace _3_Application.Dtos.DeliverySheet
{
    public class ShippingSheetDto : DeliverySheetDto
    {
        public int DriverId { get; set; }

        public int TruckId { get; set; }


        public DriverDto Driver { get; set; }

        public TruckDto Truck { get; set; }


        public ShippingSheetDto()
        {
            DeliveryMethod = DeliveryMethodEnum.Shipping;
        }
    }
}